core/src/main/java/feast/core/model/Source.java

```diff
@@ -210,11 +161,25 @@ public boolean equals(Object o) {
       return false;
     }
     Source source = (Source) o;
-    return id.equals(source.id);
+    return this.equalTo(source);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hash(id);
+    return Objects.hash(getType(), getConfig());
+  }
+
+  /**
+   * Returns the type of this Source in String format
+   *
+   * @return Source type in String format
+   */
+  public String getTypeString() {
+    return this.getType().getValueDescriptor().getName();
+  }
+
+  @Override
+  public String toString() {
```

Comment:
if this one is supposed to be used in `createJobId` then it's not exactly correct. `this.id` won't be populated in ConsolidatedSource and more stable formula (type + config) should be used as kafka consumer group id

Reply:
And I would rather have this logic in `createJobId` if that's case

Reply:
Changed to `Source.type-hash(Source.config)`